contracts/eosio.system/include/eosio.system/eosio.system.hpp

```diff
@@ -509,12 +509,38 @@ namespace eosiosystem {
       uint64_t  primary_key() const { return giftee.value; } // unique as one giftee can only hold gifted ram from one gifter
    };
 
+   struct [[eosio::table("peerkeys"), eosio::contract("eosio.system")]] peer_key
+   {
+      struct v0_data {
+         std::optional<eosio::public_key> pubkey;  // peer key for network message authentication
+         EOSLIB_SERIALIZE( v0_data, (pubkey) )
+      };
+
+      name                                           proposer_finalizer_name;
+      uint32_t                                       block_num; // block number where this row was emplaced or modified
```

Comment:
A block timestamp can serve this purpose just as well. I rather we avoid a dependency on the host function to get block number if we don't really need it.

Reply:
Just out of curiosity, why is it better to avoid a dependency on the host function? Might it be removed someday?
The timestamp is larger, right (`uint64_t`).
Also, just to be sure, when a node is processing a block produced by a different node, it uses that block's timestamp I assume, not its own clock.

Reply:
Is there a direct way to get the block timestamp in a contract? Or do I have to do a lookup into the `block_info_table`?

Reply:
`eosio::current_block_time()`

Reply:
The host function returns `uint64_t` but we only need to store a `block_timestamp` which is 32-bit

Reply:
So why is using the `current_time()` host function better that using `eosio::current_block_number()`? 
And will `current_time()` return the node time on the system clock, or the actual block timestamp?

Reply:
> So why is using the `current_time()` host function better that using `eosio::current_block_number()`?

I think the question is more like why is using `current_time()` host function better than using `get_block_num()` host function. (since really the contract should be interacting with the eosiolib wrappers in both cases). Def a question for @arhag 

> And will `current_time()` return the node time on the system clock, or the actual block timestamp?

It's the pending block time (so actual block timestamp)

Reply:
https://github.com/eosnetworkfoundation/eos-system-contracts/pull/185/commits/3db3a013a52917fa94df095934c947699a4cf7d3 and https://github.com/AntelopeIO/spring/pull/1276/commits/2829b108af330e676a8ca903fddfc5644d0467e8